integration-libs/opf/checkout/components/opf-checkout-payment-wrapper/opf-checkout-payment-wrapper.service.ts

```diff
@@ -0,0 +1,273 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import {
+  ActiveCartFacade,
+  CartAccessCodeFacade,
+} from '@spartacus/cart/base/root';
+import {
+  DEFAULT_AUTHORIZATION_ERROR_RETRIES_COUNT,
+  GlobalMessageService,
+  GlobalMessageType,
+  HttpErrorModel,
+  HttpResponseStatus,
+  RoutingService,
+  UserIdService,
+  backOff,
+  isAuthorizationError,
+} from '@spartacus/core';
+
+import {
+  OpfMetadataStoreService,
+  OpfResourceLoaderService,
+} from '@spartacus/opf/base/root';
+import { OPF_PAYMENT_AND_REVIEW_SEMANTIC_ROUTE } from '@spartacus/opf/checkout/root';
+import {
+  OpfPaymentFacade,
+  OpfRenderPaymentMethodEvent,
+  PaymentPattern,
+  PaymentSessionData,
+} from '@spartacus/opf/payment/root';
+import { OrderFacade } from '@spartacus/order/root';
+import {
+  BehaviorSubject,
+  Observable,
+  combineLatest,
+  of,
+  throwError,
+} from 'rxjs';
+import { catchError, filter, map, switchMap, take, tap } from 'rxjs/operators';
+
+@Injectable()
+export class OpfCheckoutPaymentWrapperService {
+  protected lastPaymentOptionId?: number;
+
+  protected activeCartId?: string;
+
+  protected renderPaymentMethodEvent$ =
+    new BehaviorSubject<OpfRenderPaymentMethodEvent>({
+      isLoading: false,
+      isError: false,
+    });
+
+  constructor(
+    protected opfPaymentService: OpfPaymentFacade,
```

Comment:
[MEDIUM]

```suggestion
    protected opfPaymentFacade: OpfPaymentFacade
```

To avoid ambiguity, let's use the word `Facade` in the property name, becaue we're using a lazy-loadable Facade, but not diractly its underlying Service.

NOTE: analogical comment applies also inside the class `OpfPaymentVerificationService`.

Reply:
Fixed.